src/core/net/ip6.cpp

```diff
@@ -251,7 +264,11 @@ ThreadError Ip6::HandleOptions(Message &message, Header &header, bool &forward)
         switch (optionHeader.GetType())
         {
         case OptionMpl::kType:
-            SuccessOrExit(error = mMpl.ProcessOption(message, header.GetSource(), forward));
+            if (receive)
```

Comment:
Why are we only handling the MPL Option if `receive` is `true`?  IPv6 hop-by-hop options are to be processed by any forwarding device whether or not the message is destined for it.